Source/DSPSpy/main_spy.cpp

```diff
@@ -567,7 +589,8 @@ int main()
     if (PAD_ButtonsDown(0) & PAD_BUTTON_B)
 #endif
     {
-      dsp_steps = 0;  // Let's not add the new steps after the original ones. That was just annoying.
+      dsp_steps =
+          0;  // Let's not add the new steps after the original ones. That was just annoying.
```

Comment:
The comment was first added in 11206892b73303fa5067d7ee702026f627607e76. I think before that, if you pressed B to start over, it wouldn't actually start over, but instead would append the results from rerunning it after the results from the first run (so if there were 3 results from the first run, pressing B would give 6 results, instead of clearing it and then giving 3 results).

The comment also got duplicated in 9d3eb63508727226560d4b1555873cdf596e3a07 and then changed in b3043324695231be5fb6d8d96d5a9b0f74abd110.

A better comment would probably be "Reset step counter so that the ucode starts from the beginning instead of appending to the list of steps." Perhaps we should also be setting `show_step` to 0 here?

Reply:
I've rewritten the comment using @Pokechu22's suggestion. Pokechu sounds uncertain, so I have not changed `show_step` to 0, but if anyone else agrees then I would be willing to make that change as well.